dev/tests/integration/framework/Magento/TestFramework/Helper/Amqp.php

```diff
@@ -13,26 +13,70 @@
  */
 class Amqp
 {
+    const CONFIG_PATH_HOST = 'queue/amqp/host';
+    const CONFIG_PATH_USER = 'queue/amqp/user';
+    const CONFIG_PATH_PASSWORD = 'queue/amqp/password';
+    const DEFAULT_MANAGEMENT_PORT = '15672';
+
     /**
      * @var Curl
      */
     private $curl;
 
+    /**
+     * @var \Magento\Framework\App\DeploymentConfig
+     */
+    private $deploymentConfig;
+
     /**
      * RabbitMQ API host
      *
      * @var string
      */
-    private $host = 'http://localhost:15672/api/';
+    private $host;
 
     /**
      * Initialize dependencies.
+     * @param \Magento\Framework\App\DeploymentConfig $deploymentConfig
      */
-    public function __construct()
-    {
+    public function __construct(
+        \Magento\Framework\App\DeploymentConfig $deploymentConfig
+    ) {
+        $this->deploymentConfig = $deploymentConfig ?? \Magento\TestFramework\Helper\Bootstrap::getObjectManager()
+            ->get(\Magento\Framework\App\DeploymentConfig::class);
         $this->curl = new Curl();
-        $this->curl->setCredentials('guest', 'guest');
+        $this->curl->setCredentials(
+            $this->deploymentConfig->get(self::CONFIG_PATH_USER),
+            $this->deploymentConfig->get(self::CONFIG_PATH_PASSWORD)
+        );
         $this->curl->addHeader('content-type', 'application/json');
+        $this->host = sprintf(
+            'http://%s:%s/api/',
+            $this->deploymentConfig->get(self::CONFIG_PATH_HOST),
+            defined('RABBITMQ_MANAGEMENT_PORT') ? RABBITMQ_MANAGEMENT_PORT : self::DEFAULT_MANAGEMENT_PORT
+        );
+    }
+
+    /**
+     * Check that the RabbitMQ instance has the management plugin installed and the api
+     * is available.
+     *
+     * @throws \Magento\TestFramework\MessageQueue\PreconditionFailedException
+     * @return bool
+     */
+    public function isAvailable()
```

Comment:
Please refactor this method to return true or false instead of throwing exceptions.

Reply:
@sivaschenko addressed here https://github.com/magento/magento2/pull/18978/commits/92e78c3f769a778f815ba450c4b1cf499ad5e41b